### PR TITLE
[#8] [#10] [#19] Add RDS Storage, CloudWatch logging, and Phoenix template to ECS Fargate

### PR DESCRIPTION
**Pull request by liamstevens111**
- #8 
- #10 
- #19 

## IMPORTANT: This contains ECS PR (https://github.com/liamstevens111/terraform-infrastructure-example/pull/14) as well due to merging PR into netowrking branch instead of main as it didn't update base branch

- [x] Use Nimble Elixir/Phoenix template to create and host inside the container using ECR (https://github.com/liamstevens111/terraform-elixir-repo)
- [x] Added RDS (not Aurora) for the storage / database. Entered these details inline in the ECS task definition (until I fix with parameter store)
- [x] Added Cloudwatch logging

## Insight 📝

The phoenix project repo ci/cd github action builds and uploads the docker image to ECR. DB sensitive data currently listed inline in task definition (prior to parameter store task). 

~~There was an issue with Fargate running the image because I think it was built on MAC M1.  After googling it seems you can change architecture during `docker buildx` to include `linux arm64` but this also failed. The only thing that worked was defining `runtime_platform` ECS task definition to declare this. With Github actions setup on phoenix project rep performing the build should fix these issues.~~ Now handled in the phoenix project ci/cd auto deploy to ECR

## Proof Of Work 📹

Load Balancer with associated DNS and target group:

Screenshot 2022-12-19 at 14 10 29

Going to the Load Balancer DNS shows the phoenix application running in Fargate:

Screenshot 2022-12-19 at 14 11 49

ECS Cluster and tasks:

Screenshot 2022-12-19 at 14 12 54

Cloudwatch logs:

Screenshot 2022-12-19 at 14 47 08

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -34,6 +34,19 @@ module "ecs" {
   task_count = 2
 
   alb_target_group_arn = module.network.alb_target_group_arn
-  subnets              = module.network.private_subnet_group_ids
-  security_groups      = [module.network.alb_security_group_id, module.network.ecs_security_group_id]
+  #TODO: Replace with private subnet groups when implemented
```
Comment (andyduong1920) [inline main.tf:39]: when implement what @liamstevens111 🤔 do we have any Github issue for this TODO? if not, how about

- Create a Github issue for this
- Link that issue on this comment `# TODO: Replace with private subnet groups when implemented on #xxx`

File: main.tf [hunk 2]
```diff
@@ -34,6 +34,19 @@ module "ecs" {
...
+  database_url    = var.database_url
+  secret_key_base = var.secret_key_base
+}
+
+module "rds" {
+  source                 = "./modules/db"
+  db_password            = var.db_password
+  app_name               = var.app_name
+  env_name               = var.env_name
+  subnet_ids             = module.network.public_subnet_group_ids
```
Comment (andyduong1920) [inline main.tf]: should we put RDS in a `private network` subnet

File: main.tf (hunk 1 above)
Comment (liamstevens111) [reply main.tf:39]: @andyduong1920 sorry here https://github.com/liamstevens111/terraform-infrastructure-example/issues/21

Will link inside 👍

File: main.tf (hunk 2 above)
Comment (liamstevens111) [reply main.tf]: As per above comment and https://github.com/liamstevens111/terraform-infrastructure-example/issues/21 private subnet doesn't exist right? I need a NAT Gateway

Comment (liamstevens111) [reply main.tf]: The two "private" subnets get auto linked to the main route table as they're not assigned another I think. So I need another route table and NAT gateway

File: main.tf [hunk 3]
```diff
@@ -32,7 +32,23 @@ module "ecs" {
...
+  database_url    = var.database_url
+  secret_key_base = var.secret_key_base
+}
+
+module "rds" {
+  source                 = "./modules/db"
+  db_name                = "liamdbstaging"
+  db_password            = var.db_password
+  namespace              = var.namespace
+  subnet_ids             = module.network.public_subnet_ids
```
Comment (andyduong1920) [inline main.tf:53]:
```suggestion
  # TODO: Replace with private subnet groups when implemented
  subnet_ids             = module.network.public_subnet_ids
```

Comment (liamstevens111) [reply main.tf:53]: @andyduong1920 Changed here [4a89c12](https://github.com/liamstevens111/terraform-infrastructure-example/pull/25/commits/4a89c1288521f425f95f3be0b28e086d09ff281c)

**Review by andyduong1920 — DISMISSED** [review]

**Review by andyduong1920 — APPROVED** [review]

**Comment by liamstevens111** [comment]
Incorporated into refactor repo here: https://github.com/liamstevens111/terraform-aws-infrastructure

**Pull request merged**